ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/LoadDatabase.java

```diff
@@ -73,24 +86,44 @@ Database readDbMetadata() throws SemanticException {
     return event.dbInMetadata(dbNameToLoadIn);
   }
 
+  private ReplLoadOpType getLoadDbType(String dbName) throws InvalidOperationException, HiveException {
+    Database db = context.hiveDb.getDatabase(dbName);
+    if (db == null) {
+      return ReplLoadOpType.LOAD_NEW;
+    }
+    if (isDbAlreadyBootstrapped(db)) {
+      throw new InvalidOperationException("Bootstrap REPL LOAD is not allowed on Database: " + dbName
+              + " as it was already done.");
+    }
+    if (ReplUtils.replCkptStatus(dbName, db.getParameters(), context.dumpDirectory)) {
+      return ReplLoadOpType.LOAD_SKIP;
+    }
+    if (isDbEmpty(dbName)) {
+      return ReplLoadOpType.LOAD_REPLACE;
```

Comment:
empty db is only for default ..others throw error if db exists ?

Reply:
I think, we decided to keep the behaviour same as base code. So, in base code, we allow empty db for both default and non-default cases.